### ISSUE TITLE
BUG: <Please write a comprehensive title after the 'BUG: ' prefix>

### ISSUE THREAD

**Issue by Crejan** [numpy/numpy#26096]
### Describe the issue:

High CPU load when using simple numpy matrix operation periodically
In the example i am only calculating one matrix inverse every 20ms but I have 3/8 CPU cores under full load at all times

### Reproduce the code example:

```python
import time
import numpy
import psutil

def test_numpy():
    N = 100
    data = [numpy.random.rand(4, 4) for _ in range(N)]
    cpu = 0
    for m in data:
        numpy.linalg.inv(m)
        time.sleep(0.05)
        cpu += psutil.cpu_percent()
    print(f"\ncpu core usage: {round(cpu / N * psutil.cpu_count())}%")
```

### Error message:

```shell
No error. But very high CPU load on the overall system even for small calculations
```

### Python and NumPy Versions:

1.24.4
3.11.8

### Runtime Environment:

[{'simd_extensions': {'baseline': ['SSE', 'SSE2', 'SSE3'],
                      'found': ['SSSE3',
                                'SSE41',
                                'POPCNT',
                                'SSE42',
                                'AVX',
                                'F16C',
                                'FMA3',
                                'AVX2'],
                      'not_found': ['AVX512F',
                                    'AVX512CD',
                                    'AVX512_KNL',
                                    'AVX512_KNM',
                                    'AVX512_SKX',
                                    'AVX512_CLX',
                                    'AVX512_CNL',
                                    'AVX512_ICL']}},
 {'architecture': 'Haswell',
  'filepath': '/home/jan/dev/pyurctrl/build/venv/lib/python3.11/site-packages/numpy.libs/libopenblas64_p-r0-15028c96.3.21.so',
  'internal_api': 'openblas',
  'num_threads': 8,
  'prefix': 'libopenblas',
  'threading_layer': 'pthreads',

### Context for the issue:

periodic calculations on robot telemetry are slowing down the overall system performance dramatically.

**Comment by andyfaff** [comment]
This will be caused by the overhead of the 'timing' loop you created, not the cost of a numpy calculation.

```
import numpy as np
data = np.random.rand(4, 4)
%timeit np.linalg.inv(data)
```
gives
```
2.51 µs ± 34 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```
so any extra delay/slowdown is the print, asking for cpu load, and the sleep.

**Comment by Crejan** [comment]
this is just a minimal example. in my real application i am experiencing the same while only calculating after receiving data. 
using scipy.linalg.inv i have a total cpu core load of 10-20% in this application. using numpy.linalg.inv the total cpu core load is between 280-320% and my cpu fans are spinning up

**Comment by Crejan** [comment]
![image](https://github.com/numpy/numpy/assets/45814838/df2420f3-c2f4-49aa-b7eb-146e8b0a36f2)

i can also see this behavior inside htop. without the printing and psuti.
Time.sleep should not contribute to cpu load at all

Here is the same using scipy:
![image](https://github.com/numpy/numpy/assets/45814838/bc428975-7363-4426-b9ab-2e5d8b1544e1)

**Comment by Crejan** [comment]
### `scipy.show_config('dicts')['Build Dependencies']['blas']`
```
{'name': 'OpenBLAS', 'found': True, 'version': '0.3.18', 'detection method': 'cmake', 'include directory': 'unknown', 'lib directory': 'unknown', 'openblas configuration': 'unknown', 'pc file directory': 'unknown'}
```


### `numpy.show_config('dicts')['Build Dependencies']['blas']`
yields an error: `TypeError: show() takes 0 positional arguments but 1 was given`

### `numpy.show_config()`
```
openblas64__info:
    libraries = ['openblas64_', 'openblas64_']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None)]
    runtime_library_dirs = ['/usr/local/lib']
blas_ilp64_opt_info:
    libraries = ['openblas64_', 'openblas64_']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None)]
    runtime_library_dirs = ['/usr/local/lib']
openblas64__lapack_info:
    libraries = ['openblas64_', 'openblas64_']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None), ('HAVE_LAPACKE', None)]
    runtime_library_dirs = ['/usr/local/lib']
lapack_ilp64_opt_info:
    libraries = ['openblas64_', 'openblas64_']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None), ('BLAS_SYMBOL_SUFFIX', '64_'), ('HAVE_BLAS_ILP64', None), ('HAVE_LAPACKE', None)]
    runtime_library_dirs = ['/usr/local/lib']
Supported SIMD extensions in this NumPy install:
    baseline = SSE,SSE2,SSE3
    found = SSSE3,SSE41,POPCNT,SSE42,AVX,F16C,FMA3,AVX2
    not found = AVX512F,AVX512CD,AVX512_KNL,AVX512_KNM,AVX512_SKX,AVX512_CLX,AVX512_CNL,AVX512_ICL
```

System installed BLAS version is  OpenBLAS 0.3.21

**Comment by Crejan** [comment]
my guess is that there is something wrong with the scheduling. this is `perf top` while running the test script:
![image](https://github.com/numpy/numpy/assets/45814838/c512ed78-ca04-4c2a-a871-c3b4a7e587ff)

**Comment by testhowtest** [comment]
I see it might have some similarities to #8120 and #1225, right?

**Comment by Crejan** [comment]
yes, could be the same cause. It's been some time now, but I if I remember correctly I also saw hundreds of Thread traces when trying to do a performance tracing even for a relatively short run.

**Comment by gkcng** [comment]
I just encountered the same issue with numpy 1.25.2. Apparently it is multi-threading from BLAS. The suggestion from this StackOverflow [post](https://stackoverflow.com/questions/58150251/numpy-matrix-inverse-appears-to-use-multiple-threads) worked for me, CPU reduced from 300%+ risking my macbook pro becoming a jet engine, to well within reasonable level and fan isn't even on:

`export OPENBLAS_NUM_THREADS=1 `

numpy.show_config() doesn't show anything useful, and my openblas lib don't have useful versioning filename either:
`... versions/3.10.9/lib/python3.10/site-packages/numpy/.dylibs/libopenblas64_.0.dylib`
